# Census Datasets 2011, 2016 & 2021: Clean & Transform  Datasets

Census data for each year contains population by age, sex and personal income in separate files.

Imported, cleaned, combined and exported census datasets of 2011, 2016 and 2021. Separate export for each year. 

In [143]:
import pandas as pd
import numpy as np
import os

## Import & Combine 2011 Census Datasets

### Import & Clean 2011 Population-Age Dataset

In [144]:
# set file path and import 2011 population by age dataset
path = r"/Users/patel/Documents/CF-Data Anaylst Course/portfolio_projects/mbs_analysis/datasets/"

df_census_age_2011 = pd.read_csv(
    os.path.join(path, "original_datasets/census_data/2011/2011_population_age.csv"),
    index_col=False,
    encoding="ISO-8859-1",
)
df_census_age_2011.head()

SA3  age_0-14  age_15-24  age_25-44  age_45-64  age_65-79  age_80+
0  10101     13648       7590      15585      18904       8708     3072
1  10102     11025       6762      15554      14760       4171     1239
2  10103      3641       2198       4485       5530       2264      818
3  10104     11507       6347      12033      21878      11980     4123
4  10201     30502      19699      37671      43996      20940    10297

In [145]:
df_census_age_2011.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 351 entries, 0 to 350
Data columns (total 7 columns):
 #   Column     Non-Null Count  Dtype
---  ------     --------------  -----
 0   SA3        351 non-null    int64
 1   age_0-14   351 non-null    int64
 2   age_15-24  351 non-null    int64
 3   age_25-44  351 non-null    int64
 4   age_45-64  351 non-null    int64
 5   age_65-79  351 non-null    int64
 6   age_80+    351 non-null    int64
dtypes: int64(7)
memory usage: 19.3 KB


In [146]:
# converting SA3 to be string for merges with other datasets (census and mbs)
df_census_age_2011["SA3"] = df_census_age_2011["SA3"].astype(("str"))
df_census_age_2011.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 351 entries, 0 to 350
Data columns (total 7 columns):
 #   Column     Non-Null Count  Dtype 
---  ------     --------------  ----- 
 0   SA3        351 non-null    object
 1   age_0-14   351 non-null    int64 
 2   age_15-24  351 non-null    int64 
 3   age_25-44  351 non-null    int64 
 4   age_45-64  351 non-null    int64 
 5   age_65-79  351 non-null    int64 
 6   age_80+    351 non-null    int64 
dtypes: int64(6), object(1)
memory usage: 19.3+ KB


In [147]:
df_census_age_2011.describe()

age_0-14     age_15-24     age_25-44     age_45-64     age_65-79  \
count    351.000000    351.000000    351.000000    351.000000    351.000000   
mean   11806.190883   8166.660969  17179.524217  15541.148148   6189.886040   
std     8339.142607   6068.821251  13064.111914  10225.126635   4261.153964   
min        0.000000      0.000000      0.000000      0.000000      0.000000   
25%     6461.000000   4083.500000   8002.000000   8539.500000   3072.500000   
50%     9796.000000   6702.000000  13932.000000  13716.000000   5595.000000   
75%    15842.500000  10846.000000  23419.000000  21342.500000   8696.500000   
max    39077.000000  28164.000000  87146.000000  45569.000000  22249.000000   

            age_80+  
count    351.000000  
mean    2392.233618  
std     1902.499656  
min        0.000000  
25%     1019.500000  
50%     1987.000000  
75%     3215.000000  
max    10297.000000

Total 351 SA3 areas in census population age dataset. This is correct since MBS 2013-19 contains 346 unique SA3 areas. Further checks on missing SA3 areas will be conducted before merging datasets

### Import & Clean 2011 Population Personal Income Dataset

In [148]:
# import 2011 population by income dataset

df_census_personal_income_2011 = pd.read_csv(
    os.path.join(
        path, "original_datasets/census_data/2011/2011_population_personal_income.csv"
    ),
    index_col=False,
    encoding="ISO-8859-1",
)
df_census_personal_income_2011.head()

SA3  Negative income  Nil income  $1-$199 ($1-$10,399)  \
0  10101              333        3126                  3900   
1  10102              174        2503                  2489   
2  10103               83         799                  1103   
3  10104              295        2846                  4480   
4  10201              640        8641                 10113   

   $200-$299 ($10,400-$15,599)  $300-$399 ($15,600-$20,799)  \
0                         6157                         6630   
1                         2660                         3051   
2                         1479                         1746   
3                         8533                         8105   
4                        13944                        15751   

   $400-$599 ($20,800-$31,199)  $600-$799 ($31,200-$41,599)  \
0                         7080                         5958   
1                         3978                         4239   
2                         2064                         1878   
3                         9082                         6388   
4                        17655                        13977   

   $800-$999 ($41,600-$51,999)  $1,000-$1,249 ($52,000-$64,999)  \
0                         4256                             3934   
1                         3859                             4422   
2                         1344                             1134   
3                         4243                             3191   
4                        10481                             9723   

   $1,250-$1,499 ($65,000-$77,999)  $1,500-$1,999 ($78,000-$103,999)  \
0                             2656                              3222   
1                             3528                              4495   
2                              695                               787   
3                             1679                              1965   
4                             6587                              8093   

   $2,000 or more ($104,000 or more)  Not stated  Not applicable   Total  
0                               2258        4338           13648   67500  
1                               4085        3007           11025   53511  
2                                530        1642            3641   18931  
3                               1123        4432           11507   67880  
4                               6307       10701           30502  163110

In [149]:
df_census_personal_income_2011.shape

(351, 16)

In [150]:
df_census_personal_income_2011.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 351 entries, 0 to 350
Data columns (total 16 columns):
 #   Column                             Non-Null Count  Dtype
---  ------                             --------------  -----
 0   SA3                                351 non-null    int64
 1   Negative income                    351 non-null    int64
 2   Nil income                         351 non-null    int64
 3   $1-$199 ($1-$10,399)               351 non-null    int64
 4   $200-$299 ($10,400-$15,599)        351 non-null    int64
 5   $300-$399 ($15,600-$20,799)        351 non-null    int64
 6   $400-$599 ($20,800-$31,199)        351 non-null    int64
 7   $600-$799 ($31,200-$41,599)        351 non-null    int64
 8   $800-$999 ($41,600-$51,999)        351 non-null    int64
 9   $1,000-$1,249 ($52,000-$64,999)    351 non-null    int64
 10  $1,250-$1,499 ($65,000-$77,999)    351 non-null    int64
 11  $1,500-$1,999 ($78,000-$103,999)   351 non-null    int64
 12  $2,000 or more ($104,0

In [151]:
# converting SA3 to be string for merges with other datasets (census and mbs)
df_census_personal_income_2011["SA3"] = df_census_personal_income_2011["SA3"].astype(
    ("str")
)
df_census_personal_income_2011.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 351 entries, 0 to 350
Data columns (total 16 columns):
 #   Column                             Non-Null Count  Dtype 
---  ------                             --------------  ----- 
 0   SA3                                351 non-null    object
 1   Negative income                    351 non-null    int64 
 2   Nil income                         351 non-null    int64 
 3   $1-$199 ($1-$10,399)               351 non-null    int64 
 4   $200-$299 ($10,400-$15,599)        351 non-null    int64 
 5   $300-$399 ($15,600-$20,799)        351 non-null    int64 
 6   $400-$599 ($20,800-$31,199)        351 non-null    int64 
 7   $600-$799 ($31,200-$41,599)        351 non-null    int64 
 8   $800-$999 ($41,600-$51,999)        351 non-null    int64 
 9   $1,000-$1,249 ($52,000-$64,999)    351 non-null    int64 
 10  $1,250-$1,499 ($65,000-$77,999)    351 non-null    int64 
 11  $1,500-$1,999 ($78,000-$103,999)   351 non-null    int64 
 12  $2,000 o

In [152]:
df_census_personal_income_2011.describe()

Negative income    Nil income  $1-$199 ($1-$10,399)  \
count       351.000000    351.000000            351.000000   
mean        290.849003   3744.478632           3662.404558   
std         215.583083   3165.936958           2641.206295   
min           0.000000      0.000000              0.000000   
25%         146.000000   1678.000000           1863.000000   
50%         245.000000   2844.000000           3079.000000   
75%         377.000000   4752.500000           4871.000000   
max        1211.000000  16505.000000          13276.000000   

       $200-$299 ($10,400-$15,599)  $300-$399 ($15,600-$20,799)  \
count                   351.000000                   351.000000   
mean                   5120.683761                  4888.470085   
std                    3776.026852                  3382.369116   
min                       0.000000                     0.000000   
25%                    2451.000000                  2439.500000   
50%                    4415.000000                  4464.000000   
75%                    7091.500000                  6687.000000   
max                   23608.000000                 16338.000000   

       $400-$599 ($20,800-$31,199)  $600-$799 ($31,200-$41,599)  \
count                   351.000000                   351.000000   
mean                   5715.242165                  5127.193732   
std                    3862.770369                  3568.099756   
min                       0.000000                     0.000000   
25%                    2989.000000                  2527.000000   
50%                    5119.000000                  4493.000000   
75%                    7828.000000                  6700.500000   
max                   18110.000000                 16934.000000   

       $800-$999 ($41,600-$51,999)  $1,000-$1,249 ($52,000-$64,999)  \
count                   351.000000                       351.000000   
mean                   4092.131054                      3908.675214   
std                    2917.502148                      2842.593911   
min                       0.000000                         0.000000   
25%                    2109.500000                      1919.000000   
50%                    3610.000000                      3329.000000   
75%                    5469.000000                      5286.500000   
max                   12938.000000                     13685.000000   

       $1,250-$1,499 ($65,000-$77,999)  $1,500-$1,999 ($78,000-$103,999)  \
count                       351.000000                        351.000000   
mean                       2733.763533                       3193.450142   
std                        2061.148821                       2553.227144   
min                           0.000000                          0.000000   
25%                        1292.500000                       1402.000000   
50%                        2247.000000                       2552.000000   
75%                        3795.500000                       4480.000000   
max                       11634.000000                      15848.000000   

       $2,000 or more ($104,000 or more)    Not stated  Not applicable  \
count                         351.000000    351.000000      351.000000   
mean                         3080.592593   3910.615385    11806.193732   
std                          3521.879540   2887.139860     8339.139214   
min                             0.000000      0.000000        0.000000   
25%                           871.500000   1997.500000     6461.000000   
50%                          1936.000000   3264.000000     9796.000000   
75%                          4029.500000   5182.000000    15842.500000   
max                         21922.000000  22876.000000    39077.000000   

               Total  
count     351.000000  
mean    61275.410256  
std     41775.190657  
min         0.000000  
25%     33165.500000  
50%     53373.000000  
75%     82641.000000  
max    178432.000000

In [153]:
df_census_personal_income_2011.drop(["Total"], axis=1, inplace=True)
df_census_personal_income_2011.columns

Index(['SA3', 'Negative income', 'Nil income', '$1-$199 ($1-$10,399)',
       '$200-$299 ($10,400-$15,599)', '$300-$399 ($15,600-$20,799)',
       '$400-$599 ($20,800-$31,199)', '$600-$799 ($31,200-$41,599)',
       '$800-$999 ($41,600-$51,999)', '$1,000-$1,249 ($52,000-$64,999)',
       '$1,250-$1,499 ($65,000-$77,999)', '$1,500-$1,999 ($78,000-$103,999)',
       '$2,000 or more ($104,000 or more)', 'Not stated', 'Not applicable'],
      dtype='object')

In [154]:
rename_columns = {
    "$1-$199 ($1-$10,399)": "$1-$10,399",
    "$200-$299 ($10,400-$15,599)": "$10,400-$15,599",
    "$300-$399 ($15,600-$20,799)": "$15,600-$20,799",
    "$400-$599 ($20,800-$31,199)": "$20,800-$31,199",
    "$600-$799 ($31,200-$41,599)": "$31,200-$41,599",
    "$800-$999 ($41,600-$51,999)": "$41,600-$51,999",
    "$1,000-$1,249 ($52,000-$64,999)": "$52,000-$64,999",
    "$1,250-$1,499 ($65,000-$77,999)": "$65,000-$77,999",
    "$1,500-$1,999 ($78,000-$103,999)": "$78,000-$103,999",
    "$2,000 or more ($104,000 or more)": "$104,000+",
}

df_census_personal_income_2011.rename(columns=rename_columns, inplace=True)
df_census_personal_income_2011

SA3  Negative income  Nil income  $1-$10,399  $10,400-$15,599  \
0    10101              333        3126        3900             6157   
1    10102              174        2503        2489             2660   
2    10103               83         799        1103             1479   
3    10104              295        2846        4480             8533   
4    10201              640        8641       10113            13944   
..     ...              ...         ...         ...              ...   
346  90101                3          68          27               39   
347  90102                8          35          27               49   
348  90103                0          20          15               44   
349  99797                0           0           0                0   
350  99999                0           0           0                0   

     $15,600-$20,799  $20,800-$31,199  $31,200-$41,599  $41,600-$51,999  \
0               6630             7080             5958             4256   
1               3051             3978             4239             3859   
2               1746             2064             1878             1344   
3               8105             9082             6388             4243   
4              15751            17655            13977            10481   
..               ...              ...              ...              ...   
346               38               50              109              128   
347               45               71               47               21   
348               30               27               22               22   
349                0                0                0                0   
350                0                0                0                4   

     $52,000-$64,999  $65,000-$77,999  $78,000-$103,999  $104,000+  \
0               3934             2656              3222       2258   
1               4422             3528              4495       4085   
2               1134              695               787        530   
3               3191             1679              1965       1123   
4               9723             6587              8093       6307   
..               ...              ...               ...        ...   
346              181              110               116         88   
347               24               17                23         18   
348               15               22                20         10   
349                0                0                 0          0   
350                3                3                13          0   

     Not stated  Not applicable  
0          4338           13648  
1          3007           11025  
2          1642            3641  
3          4432           11507  
4         10701           30502  
..          ...             ...  
346         834             263  
347          36             129  
348          26             103  
349           0               0  
350           3               0  

[351 rows x 15 columns]

### Import & Clean 2011 Population-Gender Dataset

In [155]:
# import 2011 population by gender dataset

df_census_gender_2011 = pd.read_csv(
    os.path.join(path, "original_datasets/census_data/2011/2011_population_sex.csv"),
    index_col=False,
    encoding="ISO-8859-1",
)
df_census_gender_2011.head()

SA3  male_pop  female_pop  total_population
0  10101     33658       33841             67500
1  10102     26852       26660             53511
2  10103      9736        9193             18931
3  10104     33368       34514             67880
4  10201     78677       84435            163110

In [156]:
df_census_gender_2011.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 351 entries, 0 to 350
Data columns (total 4 columns):
 #   Column            Non-Null Count  Dtype
---  ------            --------------  -----
 0   SA3               351 non-null    int64
 1   male_pop          351 non-null    int64
 2   female_pop        351 non-null    int64
 3   total_population  351 non-null    int64
dtypes: int64(4)
memory usage: 11.1 KB


In [157]:
df_census_gender_2011["SA3"] = df_census_gender_2011["SA3"].astype("str")
df_census_gender_2011["SA3"].info()

<class 'pandas.core.series.Series'>
RangeIndex: 351 entries, 0 to 350
Series name: SA3
Non-Null Count  Dtype 
--------------  ----- 
351 non-null    object
dtypes: object(1)
memory usage: 2.9+ KB


In [158]:
df_census_gender_2011.describe()

male_pop    female_pop  total_population
count    351.000000    351.000000        351.000000
mean   30296.333333  30979.341880      61275.410256
std    20556.012001  21249.844061      41775.190657
min        0.000000      0.000000          0.000000
25%    16518.500000  16347.500000      33165.500000
50%    26221.000000  26755.000000      53373.000000
75%    40762.000000  42135.000000      82641.000000
max    92088.000000  90795.000000     178432.000000

### Combine 2011 Age, Gender, Income Datasets

In [159]:
# check that all datasets have the same SA3 values

# extracted SA3 values in to a set (set will remove duplicates if any)
sa3_age = set(df_census_age_2011["SA3"])
sa3_income = set(df_census_personal_income_2011["SA3"])
sa3_gender = set(df_census_gender_2011["SA3"])

# checks if sets are equal. expected diff_sa3 to be False, if they don't contain the same values
diff_sa3 = sa3_age == sa3_income == sa3_gender
print(diff_sa3)

True


In [160]:
df_census_age_income_2011 = df_census_age_2011.merge(
    df_census_personal_income_2011, how="inner", on="SA3"
)
df_census_combined_2011 = df_census_age_income_2011.merge(
    df_census_gender_2011, how="inner", on="SA3"
)
df_census_combined_2011.shape

(351, 24)

In [161]:
df_census_combined_2011.info()

<class 'pandas.core.frame.DataFrame'>
Int64Index: 351 entries, 0 to 350
Data columns (total 24 columns):
 #   Column            Non-Null Count  Dtype 
---  ------            --------------  ----- 
 0   SA3               351 non-null    object
 1   age_0-14          351 non-null    int64 
 2   age_15-24         351 non-null    int64 
 3   age_25-44         351 non-null    int64 
 4   age_45-64         351 non-null    int64 
 5   age_65-79         351 non-null    int64 
 6   age_80+           351 non-null    int64 
 7   Negative income   351 non-null    int64 
 8   Nil income        351 non-null    int64 
 9   $1-$10,399        351 non-null    int64 
 10  $10,400-$15,599   351 non-null    int64 
 11  $15,600-$20,799   351 non-null    int64 
 12  $20,800-$31,199   351 non-null    int64 
 13  $31,200-$41,599   351 non-null    int64 
 14  $41,600-$51,999   351 non-null    int64 
 15  $52,000-$64,999   351 non-null    int64 
 16  $65,000-$77,999   351 non-null    int64 
 17  $78,000-$103,999

In [162]:
df_census_combined_2011.describe()

age_0-14     age_15-24     age_25-44     age_45-64     age_65-79  \
count    351.000000    351.000000    351.000000    351.000000    351.000000   
mean   11806.190883   8166.660969  17179.524217  15541.148148   6189.886040   
std     8339.142607   6068.821251  13064.111914  10225.126635   4261.153964   
min        0.000000      0.000000      0.000000      0.000000      0.000000   
25%     6461.000000   4083.500000   8002.000000   8539.500000   3072.500000   
50%     9796.000000   6702.000000  13932.000000  13716.000000   5595.000000   
75%    15842.500000  10846.000000  23419.000000  21342.500000   8696.500000   
max    39077.000000  28164.000000  87146.000000  45569.000000  22249.000000   

            age_80+  Negative income    Nil income    $1-$10,399  \
count    351.000000       351.000000    351.000000    351.000000   
mean    2392.233618       290.849003   3744.478632   3662.404558   
std     1902.499656       215.583083   3165.936958   2641.206295   
min        0.000000         0.000000      0.000000      0.000000   
25%     1019.500000       146.000000   1678.000000   1863.000000   
50%     1987.000000       245.000000   2844.000000   3079.000000   
75%     3215.000000       377.000000   4752.500000   4871.000000   
max    10297.000000      1211.000000  16505.000000  13276.000000   

       $10,400-$15,599  ...  $41,600-$51,999  $52,000-$64,999  \
count       351.000000  ...       351.000000       351.000000   
mean       5120.683761  ...      4092.131054      3908.675214   
std        3776.026852  ...      2917.502148      2842.593911   
min           0.000000  ...         0.000000         0.000000   
25%        2451.000000  ...      2109.500000      1919.000000   
50%        4415.000000  ...      3610.000000      3329.000000   
75%        7091.500000  ...      5469.000000      5286.500000   
max       23608.000000  ...     12938.000000     13685.000000   

       $65,000-$77,999  $78,000-$103,999     $104,000+    Not stated  \
count       351.000000        351.000000    351.000000    351.000000   
mean       2733.763533       3193.450142   3080.592593   3910.615385   
std        2061.148821       2553.227144   3521.879540   2887.139860   
min           0.000000          0.000000      0.000000      0.000000   
25%        1292.500000       1402.000000    871.500000   1997.500000   
50%        2247.000000       2552.000000   1936.000000   3264.000000   
75%        3795.500000       4480.000000   4029.500000   5182.000000   
max       11634.000000      15848.000000  21922.000000  22876.000000   

       Not applicable      male_pop    female_pop  total_population  
count      351.000000    351.000000    351.000000        351.000000  
mean     11806.193732  30296.333333  30979.341880      61275.410256  
std       8339.139214  20556.012001  21249.844061      41775.190657  
min          0.000000      0.000000      0.000000          0.000000  
25%       6461.000000  16518.500000  16347.500000      33165.500000  
50%       9796.000000  26221.000000  26755.000000      53373.000000  
75%      15842.500000  40762.000000  42135.000000      82641.000000  
max      39077.000000  92088.000000  90795.000000     178432.000000  

[8 rows x 23 columns]

In [163]:
df_census_combined_2011.to_pickle(
    os.path.join(path, "clean_datasets/census_data/2011_cenus_combined.pkl")
)

## Import & Combine 2016 Census Datasets

### Import & Clean 2016 Population-Age Dataset

In [164]:
# set file path and import 2016 population by age dataset

df_census_age_2016 = pd.read_csv(
    os.path.join(path, "original_datasets/census_data/2016/2016_population_age.csv"),
    index_col=False,
    encoding="ISO-8859-1",
)
df_census_age_2016.head()

SA3  age_0-14  age_15-24  age_25-44  age_45-64  age_65-79  age_80+
0  10102     11235       6877      15839      16352       5629     1395
1  10103      3275       2164       4489       5677       2830      936
2  10104     10610       6067      11644      22307      15245     4768
3  10105      6398       3988       8133       9835       5391     1822
4  10106      7255       3921       7649      10066       5398     1650

In [165]:
df_census_age_2016.tail()

SA3  age_0-14  age_15-24  age_25-44  age_45-64  age_65-79  age_80+
353  90102       116         48        139        158         68       14
354  90103        78         72        121         97         24        0
355  90104       296        110        341        577        322       91
356  99797         0          0          6          0          0        0
357  99999         9         12          8         14         14        0

In [166]:
df_census_age_2016.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 358 entries, 0 to 357
Data columns (total 7 columns):
 #   Column     Non-Null Count  Dtype
---  ------     --------------  -----
 0   SA3        358 non-null    int64
 1   age_0-14   358 non-null    int64
 2   age_15-24  358 non-null    int64
 3   age_25-44  358 non-null    int64
 4   age_45-64  358 non-null    int64
 5   age_65-79  358 non-null    int64
 6   age_80+    358 non-null    int64
dtypes: int64(7)
memory usage: 19.7 KB


In [167]:
df_census_age_2016["SA3"].value_counts()

10102    1
31402    1
39797    1
31905    1
31904    1
        ..
20607    1
20606    1
20605    1
20604    1
99999    1
Name: SA3, Length: 358, dtype: int64

In [168]:
df_census_age_2016["SA3"] = df_census_age_2016["SA3"].astype("str")
df_census_age_2016["SA3"].value_counts()

10102    1
31402    1
39797    1
31905    1
31904    1
        ..
20607    1
20606    1
20605    1
20604    1
99999    1
Name: SA3, Length: 358, dtype: int64

### Import & Clean 2016 Population Personal Income Dataset

In [169]:
# import 2016 population by income dataset

df_census_personal_income_2016 = pd.read_csv(
    os.path.join(
        path, "original_datasets/census_data/2016/2016_population_personal_income.csv"
    ),
    index_col=False,
    encoding="ISO-8859-1",
)
df_census_personal_income_2016.head()

SA3  Negative income  Nil income  $1-$149 ($1-$7,799)  \
0  10102              161        3063                 1697   
1  10103               69         839                  603   
2  10104              250        3168                 2481   
3  10105              143        1895                 1099   
4  10106              169        1757                 1186   

   $150-$299 ($7,800-$15,599)  $300-$399 ($15,600-$20,799)  \
0                        2188                         2658   
1                        1028                         1435   
2                        5055                         7702   
3                        2051                         2774   
4                        1897                         2699   

   $400-$499 ($20,800-$25,999)  $500-$649 ($26,000-$33,799)  \
0                         2732                         2657   
1                         1437                         1462   
2                         7132                         6035   
3                         2833                         2487   
4                         2605                         2432   

   $650-$799 ($33,800-$41,599)  $800-$999 ($41,600-$51,999)  \
0                         3187                         3921   
1                         1490                         1543   
2                         5404                         4999   
3                         2473                         2430   
4                         2349                         2418   

   $1,000-$1,249 ($52,000-$64,999)  $1,250-$1,499 ($65,000-$77,999)  \
0                             4450                             3639   
1                             1362                              871   
2                             4212                             2405   
3                             2360                             1429   
4                             2226                             1465   

   $1,500-$1,749 ($78,000-$90,999)  $1,750-$1,999 ($91,000-$103,999)  \
0                             3312                              2220   
1                              669                               459   
2                             1633                              1122   
3                             1219                               879   
4                             1406                               948   

   $2,000-$2,999 ($104,000-$155,999)  $3,000 or more ($156,000 or more)  \
0                               4091                               1812   
1                                561                                292   
2                               1224                                671   
3                               1055                                399   
4                               1513                                817   

   Not stated  Not applicable  Total  
0        4308           11235  57333  
1        1973            3275  19365  
2        6536           10610  70642  
3        3640            6398  35561  
4        2803            7255  35939

In [170]:
df_census_personal_income_2016.shape

(358, 19)

In [171]:
df_census_personal_income_2016.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 358 entries, 0 to 357
Data columns (total 19 columns):
 #   Column                             Non-Null Count  Dtype
---  ------                             --------------  -----
 0   SA3                                358 non-null    int64
 1   Negative income                    358 non-null    int64
 2   Nil income                         358 non-null    int64
 3   $1-$149 ($1-$7,799)                358 non-null    int64
 4   $150-$299 ($7,800-$15,599)         358 non-null    int64
 5   $300-$399 ($15,600-$20,799)        358 non-null    int64
 6   $400-$499 ($20,800-$25,999)        358 non-null    int64
 7   $500-$649 ($26,000-$33,799)        358 non-null    int64
 8   $650-$799 ($33,800-$41,599)        358 non-null    int64
 9   $800-$999 ($41,600-$51,999)        358 non-null    int64
 10  $1,000-$1,249 ($52,000-$64,999)    358 non-null    int64
 11  $1,250-$1,499 ($65,000-$77,999)    358 non-null    int64
 12  $1,500-$1,749 ($78,000

In [172]:
# converting SA3 to be string for merges with other datasets (census and mbs)
df_census_personal_income_2016["SA3"] = df_census_personal_income_2016["SA3"].astype(
    ("str")
)
df_census_personal_income_2016.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 358 entries, 0 to 357
Data columns (total 19 columns):
 #   Column                             Non-Null Count  Dtype 
---  ------                             --------------  ----- 
 0   SA3                                358 non-null    object
 1   Negative income                    358 non-null    int64 
 2   Nil income                         358 non-null    int64 
 3   $1-$149 ($1-$7,799)                358 non-null    int64 
 4   $150-$299 ($7,800-$15,599)         358 non-null    int64 
 5   $300-$399 ($15,600-$20,799)        358 non-null    int64 
 6   $400-$499 ($20,800-$25,999)        358 non-null    int64 
 7   $500-$649 ($26,000-$33,799)        358 non-null    int64 
 8   $650-$799 ($33,800-$41,599)        358 non-null    int64 
 9   $800-$999 ($41,600-$51,999)        358 non-null    int64 
 10  $1,000-$1,249 ($52,000-$64,999)    358 non-null    int64 
 11  $1,250-$1,499 ($65,000-$77,999)    358 non-null    int64 
 12  $1,500-$

In [173]:
df_census_personal_income_2016.describe()

Negative income    Nil income  $1-$149 ($1-$7,799)  \
count       358.000000    358.000000           358.000000   
mean        276.243017   4769.539106          2238.494413   
std         236.839885   4431.003748          1698.844405   
min           0.000000      0.000000             0.000000   
25%         129.000000   1929.500000          1123.000000   
50%         214.000000   3335.500000          1844.000000   
75%         351.750000   6106.500000          2915.000000   
max        1338.000000  23876.000000          7737.000000   

       $150-$299 ($7,800-$15,599)  $300-$399 ($15,600-$20,799)  \
count                  358.000000                   358.000000   
mean                  3754.502793                  4523.703911   
std                   2815.911278                  3343.960031   
min                      0.000000                     0.000000   
25%                   1855.000000                  2085.250000   
50%                   3238.000000                  3937.500000   
75%                   5052.250000                  6391.000000   
max                  16764.000000                 18964.000000   

       $400-$499 ($20,800-$25,999)  $500-$649 ($26,000-$33,799)  \
count                   358.000000                   358.000000   
mean                   4341.162011                  3981.318436   
std                    3096.220392                  2788.142379   
min                       0.000000                     0.000000   
25%                    2042.500000                  1895.250000   
50%                    3910.000000                  3572.500000   
75%                    6006.250000                  5451.500000   
max                   15022.000000                 12895.000000   

       $650-$799 ($33,800-$41,599)  $800-$999 ($41,600-$51,999)  \
count                   358.000000                   358.000000   
mean                   3984.639665                  4343.335196   
std                    2884.480722                  3226.960756   
min                       0.000000                     0.000000   
25%                    1768.000000                  2038.750000   
50%                    3489.000000                  3753.000000   
75%                    5297.250000                  5719.000000   
max                   14044.000000                 16188.000000   

       $1,000-$1,249 ($52,000-$64,999)  $1,250-$1,499 ($65,000-$77,999)  \
count                       358.000000                       358.000000   
mean                       4437.089385                      3043.907821   
std                        3324.680308                      2342.484175   
min                           0.000000                         0.000000   
25%                        2160.750000                      1359.500000   
50%                        3658.000000                      2466.000000   
75%                        5962.750000                      4289.000000   
max                       16378.000000                     12913.000000   

       $1,500-$1,749 ($78,000-$90,999)  $1,750-$1,999 ($91,000-$103,999)  \
count                       358.000000                        358.000000   
mean                       2577.860335                       1784.885475   
std                        2014.595213                       1440.788992   
min                           0.000000                          0.000000   
25%                        1138.500000                        767.000000   
50%                        2110.500000                       1419.000000   
75%                        3670.250000                       2463.000000   
max                       12585.000000                       9411.000000   

       $2,000-$2,999 ($104,000-$155,999)  $3,000 or more ($156,000 or more)  \
count                         358.000000                         358.000000   
mean                         2686.421788                        1666.256983   
std                          2507.418336                

All values are integers, no misisng or negative values.

In [174]:
# dropping total column as its not required. Total population is in population_gender file.
df_census_personal_income_2016.drop(["Total"], axis=1, inplace=True)
df_census_personal_income_2016.columns

Index(['SA3', 'Negative income', 'Nil income', '$1-$149 ($1-$7,799)',
       '$150-$299 ($7,800-$15,599)', '$300-$399 ($15,600-$20,799)',
       '$400-$499 ($20,800-$25,999)', '$500-$649 ($26,000-$33,799)',
       '$650-$799 ($33,800-$41,599)', '$800-$999 ($41,600-$51,999)',
       '$1,000-$1,249 ($52,000-$64,999)', '$1,250-$1,499 ($65,000-$77,999)',
       '$1,500-$1,749 ($78,000-$90,999)', '$1,750-$1,999 ($91,000-$103,999)',
       '$2,000-$2,999 ($104,000-$155,999)',
       '$3,000 or more ($156,000 or more)', 'Not stated', 'Not applicable'],
      dtype='object')

### Import & Clean 2016 Population-Gender Dataset

In [175]:
# import 2011 population by gender dataset

df_census_gender_2016 = pd.read_csv(
    os.path.join(path, "original_datasets/census_data/2016/2016_population_sex.csv"),
    index_col=False,
    encoding="ISO-8859-1",
)
df_census_gender_2016.head()

SA3  male_pop  female_pop  total_population
0  10102     28830       28502             57333
1  10103     10013        9356             19365
2  10104     34606       36035             70642
3  10105     17991       17571             35561
4  10106     17677       18259             35939

In [176]:
df_census_gender_2016.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 358 entries, 0 to 357
Data columns (total 4 columns):
 #   Column            Non-Null Count  Dtype
---  ------            --------------  -----
 0   SA3               358 non-null    int64
 1   male_pop          358 non-null    int64
 2   female_pop        358 non-null    int64
 3   total_population  358 non-null    int64
dtypes: int64(4)
memory usage: 11.3 KB


In [177]:
df_census_gender_2016["SA3"] = df_census_gender_2016["SA3"].astype("str")
df_census_gender_2016["SA3"].info()

<class 'pandas.core.series.Series'>
RangeIndex: 358 entries, 0 to 357
Series name: SA3
Non-Null Count  Dtype 
--------------  ----- 
358 non-null    object
dtypes: object(1)
memory usage: 2.9+ KB


In [178]:
df_census_gender_2016.describe()

male_pop     female_pop  total_population
count     358.000000     358.000000        358.000000
mean    32253.069832   33115.254190      65368.477654
std     22998.454480   23764.832765      46736.444696
min         0.000000       0.000000          0.000000
25%     16946.250000   17047.250000      33696.500000
50%     27028.000000   27617.500000      54549.000000
75%     43462.750000   44968.250000      89195.000000
max    111369.000000  110528.000000     221898.000000

### Combine 2016 Age, Gender, Income Datasets

In [179]:
# check that all datasets have the same SA3 values

# extracted SA3 values in to a set (set will remove duplicates if any)
sa3_age = set(df_census_age_2016["SA3"])
sa3_income = set(df_census_personal_income_2016["SA3"])
sa3_gender = set(df_census_gender_2016["SA3"])

# checks if sets are equal. expected diff_sa3 to be False, if they don't contain the same values
diff_sa3 = sa3_age == sa3_income == sa3_gender
print(diff_sa3)

True


In [180]:
df_census_age_income_2016 = df_census_age_2016.merge(
    df_census_personal_income_2016, how="inner", on="SA3"
)
df_census_combined_2016 = df_census_age_income_2016.merge(
    df_census_gender_2016, how="inner", on="SA3"
)
df_census_combined_2016.shape

(358, 27)

In [181]:
df_census_combined_2016.info()

<class 'pandas.core.frame.DataFrame'>
Int64Index: 358 entries, 0 to 357
Data columns (total 27 columns):
 #   Column                             Non-Null Count  Dtype 
---  ------                             --------------  ----- 
 0   SA3                                358 non-null    object
 1   age_0-14                           358 non-null    int64 
 2   age_15-24                          358 non-null    int64 
 3   age_25-44                          358 non-null    int64 
 4   age_45-64                          358 non-null    int64 
 5   age_65-79                          358 non-null    int64 
 6   age_80+                            358 non-null    int64 
 7   Negative income                    358 non-null    int64 
 8   Nil income                         358 non-null    int64 
 9   $1-$149 ($1-$7,799)                358 non-null    int64 
 10  $150-$299 ($7,800-$15,599)         358 non-null    int64 
 11  $300-$399 ($15,600-$20,799)        358 non-null    int64 
 12  $400-$49

In [182]:
df_census_combined_2016.describe()

age_0-14     age_15-24      age_25-44     age_45-64     age_65-79  \
count    358.000000    358.000000     358.000000    358.000000    358.000000   
mean   12191.516760   8347.365922   18193.726257  16365.203911   7623.905028   
std     9266.882438   6712.229282   15020.390280  11054.287733   5209.452452   
min        0.000000      0.000000       0.000000      0.000000      0.000000   
25%     6281.500000   3861.500000    8011.500000   8766.750000   3858.000000   
50%     9552.500000   6574.500000   14170.500000  14375.500000   6989.500000   
75%    16259.500000  11345.500000   25371.250000  22578.500000  10747.000000   
max    54610.000000  40357.000000  105848.000000  49083.000000  28099.000000   

            age_80+  Negative income    Nil income  $1-$149 ($1-$7,799)  \
count    358.000000       358.000000    358.000000           358.000000   
mean    2646.237430       276.243017   4769.539106          2238.494413   
std     2065.665474       236.839885   4431.003748          1698.844405   
min        0.000000         0.000000      0.000000             0.000000   
25%     1170.500000       129.000000   1929.500000          1123.000000   
50%     2267.000000       214.000000   3335.500000          1844.000000   
75%     3674.000000       351.750000   6106.500000          2915.000000   
max    10468.000000      1338.000000  23876.000000          7737.000000   

       $150-$299 ($7,800-$15,599)  ...  $1,250-$1,499 ($65,000-$77,999)  \
count                  358.000000  ...                       358.000000   
mean                  3754.502793  ...                      3043.907821   
std                   2815.911278  ...                      2342.484175   
min                      0.000000  ...                         0.000000   
25%                   1855.000000  ...                      1359.500000   
50%                   3238.000000  ...                      2466.000000   
75%                   5052.250000  ...                      4289.000000   
max                  16764.000000  ...                     12913.000000   

       $1,500-$1,749 ($78,000-$90,999)  $1,750-$1,999 ($91,000-$103,999)  \
count                       358.000000                        358.000000   
mean                       2577.860335                       1784.885475   
std                        2014.595213                       1440.788992   
min                           0.000000                          0.000000   
25%                        1138.500000                        767.000000   
50%                        2110.500000                       1419.000000   
75%                        3670.250000                       2463.000000   
max                       12585.000000                       9411.000000   

       $2,000-$2,999 ($104,000-$155,999)  $3,000 or more ($156,000 or more)  \
count                         358.000000                         358.000000   
mean                         2686.421788                        1666.256983   
std                          2507.418336                        2262.967387   
min                             0.000000                           0.000000   
25%                           899.250000                         419.750000   
50%                          1900.000000                         984.000000   
75%                          3715.500000                        1984.000000   
max                         18404.000000                       15561.000000   

         Not stated  Not applicable       male_pop     female_pop  \
count    358.000000      358.000000     358.000000     358.000000   
mean    4767.371508    12191.516760   32253.069832   33115.254190   
std     3286.956303     9266.882438   22998.454480   23764.832765   
min        0.000000        0.000000       0.000000       0.000000   
25%     2593.000000     6281.500000   16946.250000   17047.250000   
50%     4245.500000     9552.500000   27028.000000   27617.500000   
75%     6157.000000    16259.500000   43462.750000   44968

In [183]:
df_census_combined_2016.to_pickle(
    os.path.join(path, "clean_datasets/census_data/2016_cenus_combined.pkl")
)

## Import & Combine 2021 Census Datasets

### Import & Clean 2021 Population-Age Dataset

In [184]:
# set file path and import 2021 population by age dataset

df_census_age_2021 = pd.read_csv(
    os.path.join(path, "original_datasets/census_data/2021/2021_population_age.csv"),
    index_col=False,
    encoding="ISO-8859-1",
)
df_census_age_2021.head()

SA3  age_0-14  age_15-24  age_25-44  age_45-64  age_65-79  age_80+
0  10102     12643       6973      18711      17687       7006     1767
1  10103      3354       2275       5029       5915       3148     1001
2  10104     11075       6302      13133      22453      18433     5339
3  10105      6771       4065       8928      10245       6322     2071
4  10106      7401       4045       8076      10522       6198     1911

In [185]:
df_census_age_2021.tail()

SA3  age_0-14  age_15-24  age_25-44  age_45-64  age_65-79  age_80+
353  90102       130         49        147        166         80       21
354  90103        65         42         69         84         38       10
355  90104       351        186        404        705        423      117
356  99797         0          0          0          0          0        0
357  99999         0          0          4          3          0        0

In [186]:
df_census_age_2021.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 358 entries, 0 to 357
Data columns (total 7 columns):
 #   Column     Non-Null Count  Dtype
---  ------     --------------  -----
 0   SA3        358 non-null    int64
 1   age_0-14   358 non-null    int64
 2   age_15-24  358 non-null    int64
 3   age_25-44  358 non-null    int64
 4   age_45-64  358 non-null    int64
 5   age_65-79  358 non-null    int64
 6   age_80+    358 non-null    int64
dtypes: int64(7)
memory usage: 19.7 KB


In [187]:
df_census_age_2021["SA3"].value_counts()

10102    1
31402    1
39797    1
31905    1
31904    1
        ..
20607    1
20606    1
20605    1
20604    1
99999    1
Name: SA3, Length: 358, dtype: int64

In [188]:
df_census_age_2021["SA3"] = df_census_age_2021["SA3"].astype("str")
df_census_age_2021["SA3"].value_counts()

10102    1
31402    1
39797    1
31905    1
31904    1
        ..
20607    1
20606    1
20605    1
20604    1
99999    1
Name: SA3, Length: 358, dtype: int64

### Import & Clean 2021 Population Personal Income Dataset

In [189]:
# import 2021 population by income dataset

df_census_personal_income_2021 = pd.read_csv(
    os.path.join(
        path, "original_datasets/census_data/2021/2021_population_personal_income.csv"
    ),
    index_col=False,
    encoding="ISO-8859-1",
)
df_census_personal_income_2021.head()

SA3  Negative income  Nil income  $1-$149 ($1-$7,799)  \
0  10102              258        3127                 1472   
1  10103              127         911                  516   
2  10104              468        3564                 2315   
3  10105              270        2052                 1014   
4  10106              331        1922                 1021   

   $150-$299 ($7,800-$15,599)  $300-$399 ($15,600-$20,799)  \
0                        1796                         2460   
1                         679                         1243   
2                        3611                         7147   
3                        1463                         2576   
4                        1437                         2359   

   $400-$499 ($20,800-$25,999)  $500-$649 ($26,000-$33,799)  \
0                         2849                         2713   
1                         1345                         1315   
2                         7214                         6426   
3                         2938                         2500   
4                         2556                         2169   

   $650-$799 ($33,800-$41,599)  $800-$999 ($41,600-$51,999)  \
0                         2900                         3841   
1                         1474                         1623   
2                         5676                         5766   
3                         2400                         2706   
4                         2221                         2511   

   $1,000-$1,249 ($52,000-$64,999)  $1,250-$1,499 ($65,000-$77,999)  \
0                             4916                             4288   
1                             1655                             1194   
2                             5771                             3585   
3                             2947                             2030   
4                             2777                             1874   

   $1,500-$1,749 ($78,000-$90,999)  $1,750-$1,999 ($91,000-$103,999)  \
0                             4387                              3454   
1                             1018                               795   
2                             2760                              1838   
3                             1763                              1257   
4                             1834                              1414   

   $2,000-$2,999 ($104,000-$155,999)  $3,000-$3,499 ($156,000-$181,999)  \
0                               6822                               1378   
1                               1099                                234   
2                               2465                                469   
3                               1923                                319   
4                               2353                                538   

   $3,500 or more ($182,000 or more)  Not stated  Not applicable  Total  
0                               2286        3193           12643  64793  
1                                402        1722            3354  20717  
2                                846        5731           11075  76736  
3                                516        2962            6771  38403  
4                                886        2544            7401  38159

In [190]:
df_census_personal_income_2021.shape

(358, 20)

In [191]:
df_census_personal_income_2021.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 358 entries, 0 to 357
Data columns (total 20 columns):
 #   Column                             Non-Null Count  Dtype
---  ------                             --------------  -----
 0   SA3                                358 non-null    int64
 1   Negative income                    358 non-null    int64
 2   Nil income                         358 non-null    int64
 3   $1-$149 ($1-$7,799)                358 non-null    int64
 4   $150-$299 ($7,800-$15,599)         358 non-null    int64
 5   $300-$399 ($15,600-$20,799)        358 non-null    int64
 6   $400-$499 ($20,800-$25,999)        358 non-null    int64
 7   $500-$649 ($26,000-$33,799)        358 non-null    int64
 8   $650-$799 ($33,800-$41,599)        358 non-null    int64
 9   $800-$999 ($41,600-$51,999)        358 non-null    int64
 10  $1,000-$1,249 ($52,000-$64,999)    358 non-null    int64
 11  $1,250-$1,499 ($65,000-$77,999)    358 non-null    int64
 12  $1,500-$1,749 ($78,000

In [192]:
# converting SA3 to be string for merges with other datasets (census and mbs)
df_census_personal_income_2021["SA3"] = df_census_personal_income_2021["SA3"].astype(
    ("str")
)
df_census_personal_income_2021.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 358 entries, 0 to 357
Data columns (total 20 columns):
 #   Column                             Non-Null Count  Dtype 
---  ------                             --------------  ----- 
 0   SA3                                358 non-null    object
 1   Negative income                    358 non-null    int64 
 2   Nil income                         358 non-null    int64 
 3   $1-$149 ($1-$7,799)                358 non-null    int64 
 4   $150-$299 ($7,800-$15,599)         358 non-null    int64 
 5   $300-$399 ($15,600-$20,799)        358 non-null    int64 
 6   $400-$499 ($20,800-$25,999)        358 non-null    int64 
 7   $500-$649 ($26,000-$33,799)        358 non-null    int64 
 8   $650-$799 ($33,800-$41,599)        358 non-null    int64 
 9   $800-$999 ($41,600-$51,999)        358 non-null    int64 
 10  $1,000-$1,249 ($52,000-$64,999)    358 non-null    int64 
 11  $1,250-$1,499 ($65,000-$77,999)    358 non-null    int64 
 12  $1,500-$

In [193]:
df_census_personal_income_2021.describe()

Negative income    Nil income  $1-$149 ($1-$7,799)  \
count       358.000000    358.000000           358.000000   
mean        410.128492   4663.458101          1898.835196   
std         358.365204   4315.454299          1386.842498   
min           0.000000      0.000000             0.000000   
25%         182.500000   1902.250000           975.250000   
50%         325.000000   3274.500000          1659.000000   
75%         517.250000   5820.500000          2575.500000   
max        2204.000000  26787.000000          6729.000000   

       $150-$299 ($7,800-$15,599)  $300-$399 ($15,600-$20,799)  \
count                  358.000000                   358.000000   
mean                  2807.393855                  4395.575419   
std                   2096.979903                  3319.643804   
min                      0.000000                     0.000000   
25%                   1398.750000                  1991.500000   
50%                   2440.000000                  3752.000000   
75%                   3672.750000                  6067.750000   
max                  11954.000000                 20746.000000   

       $400-$499 ($20,800-$25,999)  $500-$649 ($26,000-$33,799)  \
count                   358.000000                   358.000000   
mean                   4417.972067                  4221.016760   
std                    3177.695822                  3022.077089   
min                       0.000000                     0.000000   
25%                    2151.750000                  2012.500000   
50%                    3959.000000                  3729.000000   
75%                    6215.500000                  5751.750000   
max                   15831.000000                 14272.000000   

       $650-$799 ($33,800-$41,599)  $800-$999 ($41,600-$51,999)  \
count                   358.000000                   358.000000   
mean                   4019.379888                  4694.360335   
std                    2926.346033                  3542.528195   
min                       0.000000                     0.000000   
25%                    1863.500000                  2165.750000   
50%                    3442.500000                  4006.500000   
75%                    5484.000000                  6405.750000   
max                   14949.000000                 20376.000000   

       $1,000-$1,249 ($52,000-$64,999)  $1,250-$1,499 ($65,000-$77,999)  \
count                       358.000000                       358.000000   
mean                       5214.455307                      3918.687151   
std                        3940.877698                      3017.026662   
min                           0.000000                         0.000000   
25%                        2492.750000                      1792.000000   
50%                        4469.500000                      3292.500000   
75%                        6991.250000                      5466.500000   
max                       22909.000000                     16555.000000   

       $1,500-$1,749 ($78,000-$90,999)  $1,750-$1,999 ($91,000-$103,999)  \
count                       358.000000                        358.000000   
mean                       3433.002793                       2538.002793   
std                        2639.316859                       1982.135688   
min                           0.000000                          0.000000   
25%                        1536.250000                       1088.250000   
50%                        2815.000000                       2041.500000   
75%                        4827.250000                       3571.750000   
max                       14067.000000                      11893.000000   

       $2,000-$2,999 ($104,000-$155,999)  $3,000-$3,499 ($156,000-$181,999)  \
count                         358.000000                         358.000000   
mean                         4399.527933                        1023.340782   
std                          3793.497265                

In [194]:
df_census_personal_income_2021.drop(["Total"], inplace=True, axis=1)
df_census_personal_income_2021.columns

Index(['SA3', 'Negative income', 'Nil income', '$1-$149 ($1-$7,799)',
       '$150-$299 ($7,800-$15,599)', '$300-$399 ($15,600-$20,799)',
       '$400-$499 ($20,800-$25,999)', '$500-$649 ($26,000-$33,799)',
       '$650-$799 ($33,800-$41,599)', '$800-$999 ($41,600-$51,999)',
       '$1,000-$1,249 ($52,000-$64,999)', '$1,250-$1,499 ($65,000-$77,999)',
       '$1,500-$1,749 ($78,000-$90,999)', '$1,750-$1,999 ($91,000-$103,999)',
       '$2,000-$2,999 ($104,000-$155,999)',
       '$3,000-$3,499 ($156,000-$181,999)',
       '$3,500 or more ($182,000 or more)', 'Not stated', 'Not applicable'],
      dtype='object')

### Import & Clean 2021 Population-Gender Dataset

In [195]:
# import 2021 population by gender dataset

df_census_gender_2021 = pd.read_csv(
    os.path.join(path, "original_datasets/census_data/2021/2021_population_sex.csv"),
    index_col=False,
    encoding="ISO-8859-1",
)
df_census_gender_2021.head()

SA3  male_pop  female_pop  total_population
0  10102     32664       32122             64793
1  10103     10674       10046             20717
2  10104     37632       39101             76736
3  10105     19335       19072             38403
4  10106     18787       19369             38159

In [196]:
df_census_gender_2021.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 358 entries, 0 to 357
Data columns (total 4 columns):
 #   Column            Non-Null Count  Dtype
---  ------            --------------  -----
 0   SA3               358 non-null    int64
 1   male_pop          358 non-null    int64
 2   female_pop        358 non-null    int64
 3   total_population  358 non-null    int64
dtypes: int64(4)
memory usage: 11.3 KB


In [197]:
df_census_gender_2021["SA3"] = df_census_gender_2021["SA3"].astype("str")
df_census_gender_2021["SA3"].info()

<class 'pandas.core.series.Series'>
RangeIndex: 358 entries, 0 to 357
Series name: SA3
Non-Null Count  Dtype 
--------------  ----- 
358 non-null    object
dtypes: object(1)
memory usage: 2.9+ KB


In [198]:
df_census_gender_2021.describe()

male_pop     female_pop  total_population
count     358.000000     358.000000        358.000000
mean    35042.229050   35971.013966      71013.212291
std     25608.538012   26254.400918      51836.230309
min         0.000000       0.000000          0.000000
25%     17899.000000   18410.000000      36142.500000
50%     28869.500000   29892.000000      58863.500000
75%     46866.250000   49601.000000      96576.500000
max    149962.000000  146783.000000     296748.000000

### Combine 2021 Age, Gender, Income Datasets

In [199]:
# check that all datasets have the same SA3 values

# extracted SA3 values in to a set (set will remove duplicates if any)
sa3_age = set(df_census_age_2021["SA3"])
sa3_income = set(df_census_personal_income_2021["SA3"])
sa3_gender = set(df_census_gender_2021["SA3"])

# checking number of values is as expected - 358
print(len(sa3_age))

# checks if sets are equal. expected diff_sa3 to be False, if they don't contain the same values
diff_sa3 = sa3_age == sa3_income == sa3_gender
print(diff_sa3)

358
True


In [200]:
df_census_age_income_2021 = df_census_age_2021.merge(
    df_census_personal_income_2021, how="inner", on="SA3"
)
df_census_combined_2021 = df_census_age_income_2021.merge(
    df_census_gender_2021, how="inner", on="SA3"
)
df_census_combined_2021.shape

(358, 28)

In [201]:
df_census_combined_2021.info()

<class 'pandas.core.frame.DataFrame'>
Int64Index: 358 entries, 0 to 357
Data columns (total 28 columns):
 #   Column                             Non-Null Count  Dtype 
---  ------                             --------------  ----- 
 0   SA3                                358 non-null    object
 1   age_0-14                           358 non-null    int64 
 2   age_15-24                          358 non-null    int64 
 3   age_25-44                          358 non-null    int64 
 4   age_45-64                          358 non-null    int64 
 5   age_65-79                          358 non-null    int64 
 6   age_80+                            358 non-null    int64 
 7   Negative income                    358 non-null    int64 
 8   Nil income                         358 non-null    int64 
 9   $1-$149 ($1-$7,799)                358 non-null    int64 
 10  $150-$299 ($7,800-$15,599)         358 non-null    int64 
 11  $300-$399 ($15,600-$20,799)        358 non-null    int64 
 12  $400-$49

In [202]:
df_census_combined_2021.describe()

age_0-14     age_15-24      age_25-44     age_45-64     age_65-79  \
count    358.000000    358.000000     358.000000    358.000000    358.000000   
mean   12955.231844   8484.231844   19867.153631  17477.293296   9165.231844   
std    10472.254115   6732.953512   16925.289416  12001.958004   6232.718495   
min        0.000000      0.000000       0.000000      0.000000      0.000000   
25%     6407.500000   3908.000000    8553.500000   9219.750000   4794.250000   
50%    10077.500000   6530.500000   15206.000000  15096.500000   8262.000000   
75%    17197.250000  11344.750000   27403.000000  23725.750000  12972.500000   
max    74874.000000  34675.000000  110560.000000  55602.000000  33680.000000   

            age_80+  Negative income    Nil income  $1-$149 ($1-$7,799)  \
count    358.000000       358.000000    358.000000           358.000000   
mean    3064.178771       410.128492   4663.458101          1898.835196   
std     2326.576573       358.365204   4315.454299          1386.842498   
min        0.000000         0.000000      0.000000             0.000000   
25%     1394.250000       182.500000   1902.250000           975.250000   
50%     2674.500000       325.000000   3274.500000          1659.000000   
75%     4213.500000       517.250000   5820.500000          2575.500000   
max    12294.000000      2204.000000  26787.000000          6729.000000   

       $150-$299 ($7,800-$15,599)  ...  $1,500-$1,749 ($78,000-$90,999)  \
count                  358.000000  ...                       358.000000   
mean                  2807.393855  ...                      3433.002793   
std                   2096.979903  ...                      2639.316859   
min                      0.000000  ...                         0.000000   
25%                   1398.750000  ...                      1536.250000   
50%                   2440.000000  ...                      2815.000000   
75%                   3672.750000  ...                      4827.250000   
max                  11954.000000  ...                     14067.000000   

       $1,750-$1,999 ($91,000-$103,999)  $2,000-$2,999 ($104,000-$155,999)  \
count                        358.000000                         358.000000   
mean                        2538.002793                        4399.527933   
std                         1982.135688                        3793.497265   
min                            0.000000                           0.000000   
25%                         1088.250000                        1653.000000   
50%                         2041.500000                        3272.500000   
75%                         3571.750000                        6180.250000   
max                        11893.000000                       26688.000000   

       $3,000-$3,499 ($156,000-$181,999)  $3,500 or more ($182,000 or more)  \
count                         358.000000                         358.000000   
mean                         1023.340782                        1850.969274   
std                          1046.862931                        2527.800581   
min                             0.000000                           0.000000   
25%                           291.750000                         444.750000   
50%                           690.500000                        1076.000000   
75%                          1377.500000                        2166.000000   
max                          7737.000000                       17716.000000   

         Not stated  Not applicable       male_pop     female_pop  \
count    358.000000      358.000000     358.000000     358.000000   
mean    4151.452514    12955.231844   35042.229050   35971.013966   
std     2892.789254    10472.254115   25608.538012   26254.400918   
min        0.000000        0.000000       0.000000       0.000000   
25%     2115.000000     6407.500000   17899.000000   18410.000000   
50%     3588.500000    10077.500000   28869.500000   29892.000000   
75%     5259.750000    17197.250000   46

In [203]:
df_census_combined_2021.to_pickle(
    os.path.join(path, "clean_datasets/census_data/2021_cenus_combined.pkl")
)